rust/worker/src/types/collection.rs

```diff
@@ -0,0 +1,88 @@
+use super::{Metadata, MetadataValueConversionError};
+use crate::{
+    chroma_proto,
+    errors::{ChromaError, ErrorCodes},
+};
+use thiserror::Error;
+use uuid::Uuid;
+
+#[derive(Debug, PartialEq)]
+pub(crate) struct Collection {
+    pub(crate) id: Uuid,
+    pub(crate) name: String,
+    pub(crate) topic: String,
+    pub(crate) metadata: Option<Metadata>,
+    pub(crate) dimension: Option<i32>,
+    pub(crate) tenant: String,
+    pub(crate) database: String,
+}
+
+#[derive(Error, Debug)]
+pub(crate) enum CollectionConversionError {
+    #[error("Invalid UUID")]
+    InvalidUuid,
+    #[error(transparent)]
+    MetadataValueConversionError(#[from] MetadataValueConversionError),
+}
+
+impl ChromaError for CollectionConversionError {
```

Comment:
is there a reason we don't use `impl_base_convert_error!` here?

Reply:
We don't extend it so we don't need to